bundles/org.openhab.ui/web/src/js/store/index.js

```diff
@@ -45,6 +46,9 @@ const store = new Vuex.Store({
     },
     setPagePath (state, value) {
       state.pagePath = value
+    },
+    setSearchQuery (state, value) {
+      state.searchQuery = value
     }
```

Comment:
> The search query was forgotten when the Items list reloaded or the Items list page was re-entered from an Item detail page.
This is also fixed by using Vuex to store the search query.

Using the store for this seems a little odd, tbh. It's normally for stuff that is truly global to the app, and useful to recall from anywhere.
This new addition is something that is only useful to circumvent internal f7 display issues, and applicable to one page (the items list page) only.
There's a nice `$f7.data` object available (see uses: https://github.com/search?q=repo%3Aopenhab%2Fopenhab-webui%20%24f7.data&type=code) to store transient data like this which is heavily related to Framework7, I would suggest using that instead!
I would also name it `$f7.data.lastItemSearchQuery` or something similar to avoid confusion about what it's for (as it's not applicable to all searchboxes).

Reply:
Looks nice, but unfortunately that does not work.

Reply:
It works, I just had a conceptional misthinking in my code.